### ISSUE TITLE
Add support for PHPUnit testing

### ISSUE THREAD

**Issue by bartlomiej-laczkowski** [eclipse-che/che#3987]
Add support for testing PHP scripts with the use of PHPUnit engine. To be able to create a PR with support implementation, issue [#3452](https://github.com/eclipse/che/issues/3452) has to be closed first. It would also be nice to add PHPUnit installation in different PHP stacks.

**Comment by bartlomiej-laczkowski** [comment]
Just to clarify some things, this pull request in current state provides also some general changes in testing model both on the server and IDE side. While working on this PR I have noticed that in the meantime some changes were added, which affects testing framework in general (i.e. [PR #4481](https://github.com/eclipse/che/pull/4481)). I was able to merge those changes into this PR, anyway the changes that I have made also affect testing framework in general and I think that someone from Codenvy and @davidfestal who added some recent changes in Java testing should take a look at this patch and I would be grateful for any discussion about possibility of "merging" the changes that come with this PR.

Below is the list of most important changes in testing framework that are a part of this PR:
- New DTO's for test result and related data (see new classes in org.eclipse.che.api.testing.shared.dto package)
- New data nodes for creating result tree in Test Result view (see new classes in org.eclipse.che.plugin.testing.ide.view.navigation.nodes package)
- Enhanced panels for presenting test result tree and related failures stack traces (new types of possible test result like warning, skipped test, etc.)

All of the classes and methods in testing framework internals that in my opinion should be changed/replaced with the ones provided in this PR were marked with @Deprecated annotation.

I am eager to answer all the questions related to the changes and ready to make necessary tweaks in this PR if the proposed changes are too intrusive.
